From 470e619a07c2cedfefb1fb753a1eaf84f9716e85 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?ChangZhuo=20Chen=20=28=E9=99=B3=E6=98=8C=E5=80=AC=29?=
 <czchen@debian.org>
Date: Sat, 11 Jun 2016 11:27:37 +0800
Subject: Relax taskwarrior version check

---
 src/tw.c | 21 ++++++++++++---------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/src/tw.c b/src/tw.c
index 602f7b3..4e1fb22 100644
--- a/src/tw.c
+++ b/src/tw.c
@@ -103,6 +103,10 @@ static char *task_get_version()
 static int task_check_version()
 {
 	char *ver;
+	unsigned int major;
+	unsigned int minor;
+	unsigned int patch;
+	int ret;
 
 	ver = task_get_version();
 
@@ -111,15 +115,14 @@ static int task_check_version()
 
 	log_debug("task version: %s", ver);
 
-	if (!strcmp(ver, "2.2.0")
-	    || !strcmp(ver, "2.0.0")
-	    || !strcmp(ver, "2.3.0")
-	    || !strcmp(ver, "2.4.0")
-	    || !strcmp(ver, "2.4.1")
-	    || !strcmp(ver, "2.5.0"))
-		return 1;
-	else
-		return 0;
+	ret = sscanf(ver, "%9u.%9u.%9u", &major, &minor, &patch);
+
+	if (ret == 3)
+		if (major == 2)
+			if (minor == 0 || minor >= 2)
+				return 1;
+
+	return 0;
 }
 
 static char *tw_exec(char *opts)
